### DISCUSSION TITLE
Helm release upgrade fails with a 'context deadline exceeded' error message

### DISCUSSION THREAD

**Discussion by jaepetto** [fluxcd/flux2#3970] [Q&A]
Hi,

I do have a Helm release that fails with the following events and Helm logs:
Events:
```
  Type     Reason  Age                 From             Message
  ----     ------  ----                ----             -------
  Normal   info    14m (x3 over 24m)   helm-controller  Helm upgrade has started
  Warning  error   9m6s (x3 over 19m)  helm-controller  Helm upgrade failed: context deadline exceeded

Last Helm logs:

resetting values to the chart's original version
performing update for app-helm-release
creating upgraded release for app-helm-release
waiting for release app-helm-release resources (created: 0 updated: 4  deleted: 0)
warning: Upgrade "app-helm-release" failed: context deadline exceeded
  Warning  error  9m6s (x3 over 19m)    helm-controller  reconciliation failed: Helm upgrade failed: context deadline exceeded
  Warning  error  2m51s (x16 over 29m)  helm-controller  reconciliation failed: upgrade retries exhausted
```

I am new to fluxcd and I don't know how to troubleshoot it further. Is there a way I can get more logs? How can I get more details on what's actually failing?

**Comment by MarkLFT** [comment]
@jaepetto Did you find a solution to this.  Since I updated an old flux CD to the latest version, I am receiving this problem, but only on 2 or 10 helm releases.

I have checked all the YAML files to look for differences between the working and problematic releases, and I am unable to find any other than what I expected.

**Reply by jaepetto** [reply]
@MarkLFT Unfortunately, I don't remember... But I don't think so...

**Reply by MarkLFT** [reply]
OK, thanks for responding so quickly.

**Reply by muandane** [reply]
I had this issue when I installed datashim as a helm release and it gave me the same error and it was due to one of the deployments in the helm chart don't have a lifecycle probe (liveness/readiness) so even though the chart was perfectly deployed and the deployments were up since it didn't have a health probes the controller assumed the helm chart is still deploying and errored out
